compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -939,6 +939,7 @@ pub fn expand_preparsed_format_args(
 
     let msg = "format argument must be a string literal";
     let fmt_sp = efmt.span;
+    let efmt_kind_is_lit: bool = matches!(efmt.kind, ast::ExprKind::Lit(_));
```

Comment:
why the `: bool` here?

Reply:
No particular reason. This declaration first had a different name, type, and right-hand side when I started working on this, and then I eventually converged on this.